packages/fuels-core/src/lib.rs

```diff
@@ -104,12 +105,12 @@ impl Tokenizable for bool {
     }
 }
 
-impl Tokenizable for String {
+impl Tokenizable for StringToken {
     fn from_token(token: Token) -> Result<Self, Error> {
         match token {
             Token::String(data) => Ok(data),
             other => Err(Error::InstantiationError(format!(
-                "Expected `String`, got {:?}",
+                "Expected `StringToken`, got {:?}",
```

Comment:
Actually, you expected a `Token::String` which in turn has a `StringToken` -- i.e. the reason you're in the catch-all is not because of not having a `StringToken` but due to not having a `Token::String`.